### PR TITLE
FOUNDATIONS: Profile Retreive By Id

### PR DESCRIPTION
**Pull request by rcruzfreelance**

### CODE REVIEW

File: Taarafo.Core/Brokers/Storages/IStorageBroker.Profiles.cs [hunk 1]
```diff
@@ -13,6 +14,7 @@ public partial interface IStorageBroker
     {
         ValueTask<Profile> InsertProfileAsync(Profile profile);
         IQueryable<Profile> SelectAllProfiles();
+        ValueTask<Profile> SelectProfileByIdAsync(Guid profileId);
```
Comment (ElbekDeveloper) [inline Taarafo.Core/Brokers/Storages/IStorageBroker.Profiles.cs:17]: @rcruzfreelance I think ideally adding brokers should be in separate PR to keep commit history pure. However, I don't know how critical is this.

File: Taarafo.Core/Services/Foundations/Profiles/ProfileService.Validations.cs [hunk 2]
```diff
@@ -39,6 +39,15 @@ private void ValidateProfileIsNotNull(Profile profile)
                 throw new NullProfileException();
             }
         }
+        private void ValidateStorageProfile(Profile maybeProfile, Guid profileId)
```
Comment (ElbekDeveloper) [inline Taarafo.Core/Services/Foundations/Profiles/ProfileService.Validations.cs:42]: Maybe, this method needs empty lines at the beginnig and the end

File: Taarafo.Core/Brokers/Storages/IStorageBroker.Profiles.cs (hunk 1 above)
Comment (hassanhabib) [reply Taarafo.Core/Brokers/Storages/IStorageBroker.Profiles.cs:17]: I agree

Comment (rcruzfreelance) [reply Taarafo.Core/Brokers/Storages/IStorageBroker.Profiles.cs:17]: Hello @ElbekDeveloper thanks for the comment I created a seperate pull request for the Broker https://github.com/hassanhabib/Taarafo.Core/pull/163

File: Taarafo.Core/Services/Foundations/Profiles/ProfileService.Validations.cs (hunk 2 above)
Comment (rcruzfreelance) [reply Taarafo.Core/Services/Foundations/Profiles/ProfileService.Validations.cs:43]: done

**Comment by hassanhabib** [comment]
@rcruzfreelance please fix the conflicts.

**Comment by rcruzfreelance** [comment]
merge conflict resolved
